File: 122-best-time-to-buy-and-sell-stock-ii/best-time-to-buy-and-sell-stock-ii.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int profit = 0 ;
        int cur_price = prices[0];

        for (int i = 1;i<prices.size();i++){
            if (prices[i] > cur_price){
                profit += prices[i] - cur_price;
            }
            cur_price = prices[i];
        }
        return profit;
        
    }
};

File: 122-best-time-to-buy-and-sell-stock-ii/best-time-to-buy-and-sell-stock-ii.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        cur_price = prices[0] 
        profit = 0
        for i in range(1,len(prices)):
            
            if prices[i] > cur_price:
                profit += prices[i] - cur_price
                cur_price = prices[i]

            else:
                cur_price = prices[i] 
        return profit
            
        


